### PR TITLE
Text input OLX and Markdown changes

### PR DESCRIPTION
**Pull request by lamagnifica**
## [DOC-3225](https://openedx.atlassian.net/browse/DOC-3225)

Updates text input problem Markdown and OLX to include label and description syntax and OLX reorganization.
### Reviewers

Possible roles follow. The PR submitter checks the boxes after each reviewer finishes and gives :+1:. 
- [x] Subject matter expert: @muzaffaryousaf (and team as makes sense)
- [ ] Subject matter expert: @cptvitamin 
- [ ] Doc team review (dev edit): @catong 
- [x] Product review: @sstack22 
- [ ] Partner support: 
- [ ] PM review: 

FYI: @pdesjardins @jaakana @mmacfarlane @cahrens 
### HTML Draft

http://draft-text-input-problem-component-2016-08.readthedocs.io/en/latest/exercises_tools/text_input.html
### Testing
- [ ] Ran ./run_tests.sh without warnings or errors
### Sandbox

https://studio-tnl4972.sandbox.edx.org
### Post-review
- [ ] Add a comment with the description of this change or link this PR to the next release notes task.
- [ ] Squash commits

### CODE REVIEW

**Comment by muzaffaryousaf** [comment]
@muhammad-ammar Can you also give it a look, since you worked on this.

File: en_us/shared/exercises_tools/text_input.rst [hunk 1]
```diff
@@ -17,155 +25,214 @@ Overview
...
+
+    = Nanjing Higher Normal Institute
+    or= National Central University
+    or= Nanjing University
+
+    [explanation]
+    Nanjing Higher Normal Institute first admitted female students in 1920.
+    [explanation]
+
+The OLX (open learning XML) markup for this example text input problem follows.
```
Comment (muhammad-ammar) [inline en_us/shared/exercises_tools/text_input.rst:60]: @lamagnifica Should `open learning` initials be in upper case?

File: en_us/shared/exercises_tools/text_input.rst [hunk 2]
```diff
@@ -17,155 +25,214 @@ Overview
...
+      <additional_answer answer="Nanjing University"/>
+      <textline size="20"/>
+    <solution>
+      <div class="detailed-solution">
+        <p>Explanation</p>
+        <p>Nanjing Higher Normal Institute first admitted female students in  1920.</p>
+      </div>
+    </solution>
+  </stringresponse>
+  </problem>
```
Comment (muhammad-ammar) [inline en_us/shared/exercises_tools/text_input.rst:80]: @lamagnifica Please indent the the XML.

File: en_us/shared/exercises_tools/text_input.rst [hunk 3]
```diff
@@ -17,155 +25,214 @@ Overview
...
+    Nanjing Higher Normal Institute first admitted female students in 1920.
+    [explanation]
+
+The OLX (open learning XML) markup for this example text input problem follows.
+
+.. code-block:: xml
+
+  <problem>
+  <stringresponse answer="Nanjing Higher Normal Institute" type="ci" >
+    <label>What was the first post-secondary school in China to allow both male and female students?</label>
```
Comment (muhammad-ammar) [inline en_us/shared/exercises_tools/text_input.rst]: @lamagnifica Should we also add `<description>` here?

File: en_us/shared/exercises_tools/text_input.rst [hunk 4]
```diff
@@ -185,86 +252,78 @@ common incorrect answers.
...
-        Hint for incorrect answer.
-      </stringequalhint>
+  <problem>
+    <stringresponse answer="Correct Answer" type="ci">
+    <label>Question text</label>
+      <correcthint>Feedback for the correct answer</correcthint>
+      <stringequalhint answer="Incorrect Answer">Hint for the incorrect answer</stringequalhint>
+      <textline size="20"/>
     </stringresponse>
+  </problem>
```
Comment (muhammad-ammar) [inline en_us/shared/exercises_tools/text_input.rst:241]: @lamagnifica Please indent the XML.

File: en_us/shared/exercises_tools/text_input.rst [hunk 5]
```diff
@@ -185,86 +252,78 @@ common incorrect answers.
...
-      <textline label="What is the largest state in the U.S. in terms of land
-        area?" size="20"/>
+    <stringresponse answer="Alaska" type="ci">
+    <label>What is the largest state in the U.S. in terms of land area?</label>
+      <correcthint>Alaska is the largest state in the U.S. in terms of not only land area, but also total area and water area. Alaska is 576,400 square miles, more than double the land area of the second largest state, Texas.</correcthint>
+      <stringequalhint answer="Texas">While many people think Texas is the largest state in terms of land area, it is actually the second largest and contains 261,797 square miles.</stringequalhint>
+      <stringequalhint answer="California">California is the third largest state and contains 155,959 square miles.</stringequalhint>
+      <textline size="20"/>
     </stringresponse>
   </problem>
```
Comment (muhammad-ammar) [inline en_us/shared/exercises_tools/text_input.rst:262]: @lamagnifica Please indent the XML.

File: en_us/shared/exercises_tools/text_input.rst [hunk 6]
```diff
@@ -484,45 +533,37 @@ Template
...
+        <additional_answer answer="Correct answer 2"/>
+        <additional_answer answer="Correct answer 3"/>
+        <stringequalhint answer="Incorrect answer 1">Provides feedback when learners submit the specified incorrect response.</stringequalhint>
+        <stringequalhint answer="Incorrect answer 2">Provides feedback when learners submit the specified incorrect response.</stringequalhint>
+      <textline size="20"/>
+    </stringresponse>
+    <demandhint>
+      <hint>The first text string to display when learners request a hint.</hint>
+      <hint>The second text string to display when learners request a hint.</hint>
+    </demandhint>
```
Comment (muhammad-ammar) [inline en_us/shared/exercises_tools/text_input.rst:535]: @lamagnifica Please indent the XML.

File: en_us/shared/exercises_tools/text_input.rst [hunk 7]
```diff
@@ -664,31 +710,30 @@ documented above.
...
+        <hintpart on="hintB">
+          <startouttext />Text of hint for incorrect answer B<endouttext />
+        </hintpart >
+      <stringhint answer="Incorrect answer C" type="ci" name="hintC" />
+        <hintpart on="hintC">
+          <startouttext />Text of hint for incorrect answer C<endouttext />
+        </hintpart >
+    </hintgroup>
+    </stringresponse>
+    <solution>
```
Comment (muhammad-ammar) [inline en_us/shared/exercises_tools/text_input.rst:813]: @lamagnifica Please indent the XML.

File: en_us/shared/exercises_tools/text_input.rst [hunk 8]
```diff
@@ -664,31 +710,30 @@ documented above.
...
-                    <startouttext />Text of hint for incorrect answer C<endouttext />
-                </hintpart >
-          </hintgroup>
-      </stringresponse>
-      <solution>
+    <stringresponse answer="Correct answer 1" type="ci regexp">
+      <label>Question text</label>
+      <additional_answer>Correct answer 2</additional_answer>
+      <additional_answer>Correct answer 3</additional_answer>
+      <textline size="20" label="label text"/>
```
Comment (muhammad-ammar) [inline en_us/shared/exercises_tools/text_input.rst]: @lamagnifica Please remove the `label` attribute. Its not supported now.

**Comment by muhammad-ammar** [comment]
@lamagnifica This is looking good. I have few comments though.

File: en_us/shared/exercises_tools/text_input.rst (hunk 1 above)
Comment (lamagnifica) [inline en_us/shared/exercises_tools/text_input.rst:60]: @muhammad-ammar I thought that we (doc team) had decided not to capitalize product names that are not trademarks, but I see that my colleague @pdesjardins has asked the same question in another one of these PRs... I will investgate! thanks.

File: en_us/shared/exercises_tools/text_input.rst (hunk 3 above)
Comment (lamagnifica) [inline en_us/shared/exercises_tools/text_input.rst]: added!

**Comment by lamagnifica** [comment]
Up to date with all edits received to date.

**Comment by sstack22** [comment]
👍  from me!

File: en_us/shared/exercises_tools/text_input.rst [hunk 9]
```diff
@@ -185,86 +222,84 @@ common incorrect answers.
...
+        <correcthint>Alaska is the largest state in the U.S. in terms of not
+         only land area, but also total area and water area. Alaska is 576,400
+         square miles, more than double the land area of the second largest
+         state, Texas.</correcthint>
+        <stringequalhint answer="Texas">While many people think Texas is the
+         largest state in terms of land area, it is actually the second
+         largest and contains 261,797 square miles.</stringequalhint>
+        <stringequalhint answer="California">California is the third largest
+         state and contains 155,959 square miles.</stringequalhint>
+        <textline size="20"/>
```
Comment (muzaffaryousaf) [inline en_us/shared/exercises_tools/text_input.rst]: Indentation a bit off for label and rest of the tags.

**Comment by muzaffaryousaf** [comment]
👍 LGTM except one NIT.

**Pull request merged**